[WIP] console-operator based on client-go

benjaminapetersen opened the pull request:
Renamed all occurance of "example" to "console". Regenerate pkg/generated/

benjaminapetersen committed:
Renamed all occurance of "example" to "console". Regenerate pkg/generated/

benjaminapetersen committed:
Update manifests to match existing console-operator manifests

benjaminapetersen committed:
Update Dockerfile & create Dockerfile.osx for local

benjaminapetersen commented:
@enj for all the review fun :)

benjaminapetersen committed:
Adding back examples/config.yaml as this will likely be needed

benjaminapetersen committed:
Update README to include expected flags & build/run process

benjaminapetersen committed:
Add `console version` command

benjaminapetersen committed:
Add some dev_*.sh scripts

enj commented on README.md:
````diff
@@ -1,17 +1,84 @@
 
-```shell
-cd "${GOPATH}/src/github.com/enj/example-operator"
+An operator for OpenShift Console built using the operator-sdk.
````
?

enj commented on dev_delete_all.sh:
```diff
@@ -0,0 +1,49 @@
+
+FILE=examples/cr.yaml
+echo "creating ${FILE}"
+oc create -f $FILE
```
delete??

benjaminapetersen replied:
quick copy paste, haven't tested that one yet.
obviously :)

enj commented on pkg/cmd/version/cmd.go:
```diff
@@ -0,0 +1,25 @@
+	String = fmt.Sprintf("ConsoleOperator %s", Raw)
+)
+
+func NewVersion() *cobra.Command {
```
This is not useful - you need `git` info like `pkg/console/version/version.go`

benjaminapetersen replied:
Agree, quick stub in, ill have to look into how that is done & how it maps to our releases since we want things formatted specifically.

enj commented on pkg/console/starter/starter.go:
```diff
@@ -0,0 +1,108 @@
+
+	// only watch a specific resource name
+	tweakListOptions := func(options *v1.ListOptions) {
+		options.FieldSelector = fields.OneTermEqualSelector("metadata.name", operator.ResourceName).String()
```
you will need to figure out if you can actually only watch a single name since you had multiple object names

benjaminapetersen replied:
Right, or if i need more than one `tweakListOptionsFor____`

enj replied:
You would need to explicitly create different informers since there is no `OR` in field selectors.  It would likely be simpler to filter at the queue.

benjaminapetersen committed:
Update README

benjaminapetersen committed:
Add dev_delete_all.sh hack script

benjaminapetersen committed:
Comment on dev_run_local.sh

benjaminapetersen committed:
Add todo to version command

benjaminapetersen committed:
Add comment to cmd/operator/cmd.go

benjaminapetersen committed:
Update dev_run_local.sh (f)

benjaminapetersen committed:
Create ClusterOperatorStatus in starter.go

benjaminapetersen commented on pkg/console/starter/starter.go:
```diff
@@ -68,41 +105,41 @@ func RunOperator(clientConfig *rest.Config, stopCh <-chan struct{}) error {
+}
+
+func (p *operatorStatusProvider) CurrentStatus() (operatorv1alpha1.OperatorStatus, error) {
+	instance, err := p.informers.Console().V1alpha1().Consoles().Lister().Consoles(operator.TargetNamespace).Get(operator.ResourceName)
```
seems to work...

benjaminapetersen committed:
Add all clients, informers to starter, operator

benjaminapetersen committed:
Create default console if CR does not exist

benjaminapetersen committed:
Update sync loop, add Apply for all subresources

benjaminapetersen committed:
Add dev_* scripts for convenience

benjaminapetersen committed:
Add TODO to types.go to describe future custom host work

benjaminapetersen committed:
Update sync loop, tmp remove OwnerRef

benjaminapetersen committed:
Return to functioning reconcile loop

- Temporarily drop ApplyRoute() to more carefully GetOrCreate route
	- ensures we don't stomp on the Spec.host
- Update Secret to manually create encoded secret
	- ApplySecret() from library-go ignores Spec.StringData
		- PR this to library-go in future
- Stub in func for CR.Status via Sync loop
	- will be a follow-up PR
- Stub in some specific status fields on Cr.Status
	- will be a follow-up PR

benjaminapetersen committed:
Handle ManagementState: Removed, refactor 'Default' resource functions to extract 'Stub'

benjaminapetersen committed:
Clean up excessive comments

benjaminapetersen committed:
f
